### DISCUSSION TITLE
Make the TransactorNode (Source Node) a standard part of every Transaction

### DISCUSSION THREAD

**Discussion by oO** [deso-protocol/dips#232] [Medium CIPs]
Currently, only the SUBMIT_POST transactions support assigning the Origin Node, and even that is done using ExtraData, which IMHO should not be used for universally useful features this early in the evolution of the DeSo blockchain. Since every single transaction must originate from a specific node ( even the BLOCK_REWARD coming from the mining node ) It would make sense for that information to be part of the base Transaction fields, not extra data:

Current Implementation: ( Submit Posts only )
```
"SubmitPostTxindexMetadata": {
    "PostHashBeingModifiedHex": "29d3177537a467f2f515a5bc0f02d64a31703ad4e94fcd28de19485774940b55",
    "ParentPostHashHex": ""
  }
},
"ExtraData": {
  "Node": "2"
}
```

Proposed Implementation: ( Every single Transaction )
```
"TransactionMetadata": {
        "BlockHashHex": "00000000000047af6e89b67416c3338dd5e6dbb8f48d2c63b7c688d6de69cc6a",
        "TxnIndexInBlock": 6,
        "TxnType": "SUBMIT_POST",
        "TransactorPublicKeyBase58Check": "BC1YLi2CCqL4ptq9zBZP9XFSJfrupjvNwFHQu4g5Nc6y5WckhSZWmNm",
        "TransactorNode": {
                "NodeId": "6",
                "NodeName": "tijn's club",
                "NodeUrl": "https://tijn.club",
                "NodeOwner: "tijn",
        } ...
```
Note that I've de-normalized the Node information but obviously the transaction itself only contains the NodeID ( which should not be an index, but something more robust and decentralized )

**Comment by darian-parrish** [comment]
I think this is a great idea and adds minimal data(nothing extra to post Txn and just single NodeID to each Txn). I also agree best to do this as a base Txn field.

This seems like an easy win - any major drawbacks here other than maybe the processing to de-normalize and additional API payload(which could be a request parameter)?

**Comment by oO** [comment]
Note that denormalizing is extra sugar to make the front end development easier. It's absolutely not necessary since you can just do that on the front end. But we already do that in a number of places ( like having the full User Profile rerturned with posts instead of just the PublicKey, so that seems to make sense in that context.

Gotta think of the @Brootles of the world 😎
